source/extensions/network/dns_resolver/apple/apple_dns_impl.cc

```diff
@@ -72,7 +72,7 @@ AppleDnsResolverImpl::startResolution(const std::string& dns_name,
     ENVOY_LOG_EVENT(debug, "apple_dns_immediate_resolution",
                     "DNS resolver resolved ({}) to ({}) without issuing call to Apple API",
                     dns_name, address->asString());
-    callback(DnsResolver::ResolutionStatus::Success,
+    callback(DnsResolver::ResolutionStatus::Success, "apple_dns_success",
```

Comment:
`apple_dns_success` wouldn't get used right?  so maybe `details` should be an `optional<string>`?

Reply:
it could - some of the logs log resolve status on success or failure.
arguably we could skip the string copy but then we'd have to do value checks everywhere so I'm not sure it's worth the trade-off

Reply:
what you say makes sense